client-libraries/rust/modality-ingest-client/src/types.rs

```diff
@@ -265,6 +312,32 @@ impl Encode for AttrVal {
     }
 }
 
+impl FromStr for AttrVal {
+    type Err = std::convert::Infallible;
+
+    fn from_str(s: &str) -> Result<Self, Self::Err> {
+        // N.B. Eventually will have to do more advanced deserialization here
+        // as our AttrVals support more interesting variants.
+        Ok(if let Ok(v) = s.to_lowercase().parse::<bool>() {
+            v.into()
+        } else if let Ok(v) = s.parse::<i64>() {
+            v.into()
+        } else if let Ok(v) = s.parse::<i128>() {
+            v.into()
+        } else if let Ok(v) = s.parse::<f64>() {
+            v.into()
+        } else if let Ok(v) = s.parse::<Nanoseconds>() {
```

Comment:
Currently this isn't going to match anything not already caught by `i128`. Can take out to avoid wasted cycles.